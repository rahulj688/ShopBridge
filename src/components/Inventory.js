import React, { Component } from 'react';
import Table from 'react-bootstrap/Table';
// import { Table } from 'semantic-ui-react';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import { RiDeleteBin6Line } from 'react-icons/ri';
import { Redirect } from 'react-router-dom';
import image1 from '../assets/image1.jpg';
import image2 from '../assets/image2.jpg';
import image3 from '../assets/image3.jpeg';
import image4 from '../assets/image4.jpg';
import image5 from '../assets/image5.jpg';

export default class Inventory extends React.Component {
  constructor(props) {
    super(props);
    // let currentInventory = this.state.inventoryProducts;
    // let inventory = [
    //   {
    //     name: 'Highlander White Slim Fit Casual Shirt',
    //     description: 'White casual shirt, has a spread collar, a full button placket, a patch pocket, long sleeves with roll-up button tabs, a curved hemline, 100% Cotton, Machine Wash, Long Sleeves, Slim Fit',
    //     price: '800',
    //     image: image1
    //   },
    //   {
    //     name: 'Men Red & Black Regular Fit Checked Casual Shirt',
    //     description: 'Red and black checked casual shirt, has a spread collar, button placket, 1 pocket, long sleeves, curved hem, 100% cotton, Machine-wash, Long Sleeves',
    //     price: '1000',
    //     image: image2
    //   },
    //   {
    //     name: 'Men Black Slim Fit Solid Casual Shirt',
    //     description: 'Black solid casual shirt, has a spread collar, long sleeves, button placket, curved hem, 1 patch pocket, Long Sleeves, Material: 65% cotton 35% linen, Machine Wash',
    //     price: '900',
    //     image: image3
    //   },
    //   {
    //     name: 'Men Yellow Printed Round Neck T-Shirt',
    //     description: 'gym-to-street look, 100% cotton, Machine-wash',
    //     price: '750',
    //     image: image4
    //   },
    //   {
    //     name: 'Men Navy Blue Tapered Fit Mid-Rise Clean Look Stretchable Jeans',
    //     description: 'Navy Blue dark wash 5-pocket mid-rise jeans, clean look, light fade, has a button and zip closure, waistband with belt loops, 98% Cotton 2% Elastane, Machine-wash',
    //     price: '1300',
    //     image: image5
    //   }
    // ]
    // if (this.props.productDetail.length > 0) {
    //   this.props.productDetail.map((obj, i) => {
    //     inventory.push(obj);
    //   })
    // }
    this.state = {
      inventoryProducts: [],
      deletedPrdoucts: []
    }
  }

  componentWillMount = () => {
    fetch('https://5f44abf43fb92f0016753a78.mockapi.io/products',
      {
        method: 'GET'
      })
      .then(async (res) => {
        let response = await res.json();
        console.log("res: ", response)
        this.setState({ inventoryProducts: response })
      })
      .catch((error) => {
      })
  }

  navigateToProductDetail = (product) => {
    fetch(`https://5f44abf43fb92f0016753a78.mockapi.io/products/${product.id}`,
      {
        method: 'GET'
      })
      .then(async (res) => {
        let response = await res.json();
        console.log("res: ", response)
      })
      .catch((error) => {
      })
    this.setState({
      navigate: true,
      product: product
    })
  }

  rows = () => {
    const rowsOutput = this.state.inventoryProducts.map((item, id) => {
      return (
        <tr key={id}>
          {/* <td><input type="radio"></input></td> */}
          <td><div onClick={() => this.navigateToProductDetail(item)}>{id + 1}</div></td>
          <td><img src={item.image} height="100" weight="100" alt="Image Not Available" /></td>
          <td>{item.name}</td>
          <td>{item.price}</td>
          <td>
            <Button variant="danger" onClick={() => { this.handleDelete(id, item) }} ><RiDeleteBin6Line /></Button>
          </td>
        </tr>
      );
    });
    return rowsOutput;
  }

  handleDelete = (id, product) => {
    fetch(`https://5f44abf43fb92f0016753a78.mockapi.io/products/${product.id}`,
      {
        method: 'DELETE'
      })
      .then(async (res) => {
        let response = await res.json();
        console.log("res: ", response)
      })
      .catch((error) => {
      })
    let rowsS = [...this.state.inventoryProducts];
    rowsS.splice(id, 1);
    this.setState({
      inventoryProducts: rowsS
    });
    let del = this.state.deletedPrdoucts;
    del.push(product);
    this.setState({ deletedPrdoucts: del });
    console.log(this.state.deletedPrdoucts)
  }

  deleteRows = () => {
    if (this.state.deletedPrdoucts) {
      const rowsOutput = this.state.deletedPrdoucts.map((item, id) => {
        return (
          <tr key={id} style={{textAlign: 'center'}}>
            {/* <td><input type="radio"></input></td> */}
            <td>{id + 1}</td>
            <td><img src={item.image} height="100" weight="100" alt="Image Not Available" /></td>
            <td>{item.name}</td>
            <td>{item.price}</td>
            <td>
              <Button variant="primary" onClick={() => { this.restore(id, item) }} >Restore</Button>
            </td>
          </tr>
        );
      });
      return rowsOutput;
    }
  }

  restore = (id, item) => {
    let rowsS = [...this.state.inventoryProducts];
    rowsS.push(item);
    let del = [...this.state.deletedPrdoucts];
    del.splice(id, 1);
    this.setState({
      inventoryProducts: rowsS,
      deletedPrdoucts: del,
      showModal: false
    });
    fetch('https://5f44abf43fb92f0016753a78.mockapi.io/products',
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(item)
      })
      .then((data) => {
        console.log('submit: ', data)
      })
      .catch((error) => {
        console.log('err: ', error)
      })
  }

  restoreAllProducts = () => {
    let productsAvailable = this.state.inventoryProducts;
    this.state.deletedPrdoucts.map((obj, index) => {
      productsAvailable.push(obj);
      fetch('https://5f44abf43fb92f0016753a78.mockapi.io/products',
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(obj)
      })
      .then((data) => {
        console.log('submit: ', data)
      })
      .catch((error) => {
        console.log('err: ', error)
      })
    })
    this.setState({
      inventoryProducts: productsAvailable,
      showModal: false,
      deletedPrdoucts: []
    })
  }

  render() {

    if (this.state.navigate) {
      return (<Redirect to={{
        pathname: "/productdetail",
        state: {
          product: this.state.product
        }
      }} />)
    }
    if (this.props.productDetail) {
      this.state.inventoryProducts.push(this.props.productDetail)
    }
    return (
      <div style={{ textAlign: 'center', marginTop: 80 }}>
        <div style={{ display: 'flex', flexDirection: 'row' }}>
          <h4 style={{ textAlign: 'left' }}>Inventory Products</h4>
          {
            this.state.deletedPrdoucts.length > 0 &&
            <Button variant='outline-info' style={{ marginLeft: 985 }} onClick={() => this.setState({ showModal: true })}>Recently Deleted</Button>
          }
        </div>
        <Table striped bordered hover style={{ marginTop: 20 }}>
          <thead>
            <tr>
              <th>Product ID</th>
              <th>Product Image</th>
              <th>Product Name</th>
              <th>Product Price</th>
              <th>Delete</th>
            </tr>
          </thead>
          <tbody>{this.rows()}</tbody>
        </Table >
        <Modal show={this.state.showModal} onHide={() => this.setState({ showModal: false })} size="lg">
          <Modal.Header closeButton>
            <Modal.Title>Modal heading</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Table striped bordered hover style={{ marginTop: 20 }}>
              <thead>
                <tr>
                  <th>Product ID</th>
                  <th>Product Image</th>
                  <th>Product Name</th>
                  <th>Product Price</th>
                  <th>Delete</th>
                </tr>
              </thead>
              <tbody>{this.deleteRows()}</tbody>
            </Table >
            <Button variant="primary" onClick={() => this.restoreAllProducts()}>Restore All</Button>
          </Modal.Body>
        </Modal>
      </div>
    )
  }
}
