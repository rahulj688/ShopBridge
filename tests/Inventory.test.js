import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Inventory from '../src/components/Inventory';
 
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Inventory />, div);
    ReactDOM.unmountComponentAtNode(div);
});
it('renders correctly', () => {
  const h = renderer.create(<Inventory />).toJSON();
  expect(h).toMatchSnapshot();
});
