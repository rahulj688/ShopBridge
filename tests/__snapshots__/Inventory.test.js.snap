exports[`test renders correctly 1`] = `
<div
  style={
    Object {
      "marginTop": 80,
      "textAlign": "center",
    }
  }>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }>
    <h4
      style={
        Object {
          "textAlign": "left",
        }
      }>
      Inventory Products
    </h4>
  </div>
  <table
    className="table table-striped table-bordered table-hover"
    style={
      Object {
        "marginTop": 20,
      }
    }>
    <thead>
      <tr>
        <th>
          Product ID
        </th>
        <th>
          Product Image
        </th>
        <th>
          Product Name
        </th>
        <th>
          Product Price
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div
            onClick={[Function]}>
            1
          </div>
        </td>
        <td>
          <img
            alt="image"
            height="100"
            src="image1.jpg"
            weight="100" />
        </td>
        <td>
          Highlander White Slim Fit Casual Shirt
        </td>
        <td>
          800
        </td>
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button">
            <svg
              className={undefined}
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg">
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none" />
                <path
                  d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
              </g>
            </svg>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            onClick={[Function]}>
            2
          </div>
        </td>
        <td>
          <img
            alt="image"
            height="100"
            src="image2.jpg"
            weight="100" />
        </td>
        <td>
          Men Red & Black Regular Fit Checked Casual Shirt
        </td>
        <td>
          1000
        </td>
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button">
            <svg
              className={undefined}
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg">
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none" />
                <path
                  d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
              </g>
            </svg>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            onClick={[Function]}>
            3
          </div>
        </td>
        <td>
          <img
            alt="image"
            height="100"
            src="image3.jpeg"
            weight="100" />
        </td>
        <td>
          Men Black Slim Fit Solid Casual Shirt
        </td>
        <td>
          900
        </td>
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button">
            <svg
              className={undefined}
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg">
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none" />
                <path
                  d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
              </g>
            </svg>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            onClick={[Function]}>
            4
          </div>
        </td>
        <td>
          <img
            alt="image"
            height="100"
            src="image4.jpg"
            weight="100" />
        </td>
        <td>
          Men Yellow Printed Round Neck T-Shirt
        </td>
        <td>
          750
        </td>
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button">
            <svg
              className={undefined}
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg">
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none" />
                <path
                  d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
              </g>
            </svg>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            onClick={[Function]}>
            5
          </div>
        </td>
        <td>
          <img
            alt="image"
            height="100"
            src="image5.jpg"
            weight="100" />
        </td>
        <td>
          Men Navy Blue Tapered Fit Mid-Rise Clean Look Stretchable Jeans
        </td>
        <td>
          1300
        </td>
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button">
            <svg
              className={undefined}
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg">
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none" />
                <path
                  d="M7 4V2h10v2h5v2h-2v15a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6H2V4h5zM6 6v14h12V6H6zm3 3h2v8H9V9zm4 0h2v8h-2V9z" />
              </g>
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
